### ISSUE TITLE
Loading error caused by the absence of stand data

### ISSUE THREAD

**Issue by shnarazk** [uiua-lang/uiua#221]
Hi,
`cargo test` emits the following error on some platform.

```
  &i example:
  Dub ← &i "example.ua" "Double"
  Dub 5
  thread 'primitive::tests::prim_docs' panicked at 'range start index 18446743840898112768 out of range for slice of length 9283584', src/sta
  nd.rs:60:53
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  thread 'primitive::tests::prim_docs' panicked at '
  Example failed:
  Dub ← &i "example.ua" "Double"
  Dub 5
  Error: The interpreter has crashed!
    Hooray! You found a bug!
    Please report this at http://github.com/uiua-lang/uiua/issues/new

    code:
    1:7
    Dub ← &i "example.ua" "Double"
    Dub 5
    at 1:7
  1 | Dub ← &i "example.ua" "Double"
            ──', src/primitive/mod.rs:1122:33


  failures:
      primitive::tests::prim_docs
```

More details by `dbg!`:

```
  [src/stand.rs:58] bytes.len() = 9275632
  [src/stand.rs:60] files_len = 59602104942712
  [src/stand.rs:62] start = 18446684471613884528
```

This problem seems to be caused by reading stand_data even if the executable doesn't include stand data, because `build_exe` wasn't called in this case.
So I think we need some checks. The following is an idea that seems to work.

```diff
diff --git a/src/stand.rs b/src/stand.rs
index c3a58a7..e26caca 100644
--- a/src/stand.rs
+++ b/src/stand.rs
@@ -7,6 +7,8 @@ use std::{
 use once_cell::sync::Lazy;
 use serde::*;
                                                                                                                            
+const STAND_DATA_SIGNATURE: &[u8; 8] = b"standsig";
+
 #[derive(Serialize, Deserialize)]
 pub struct StandFiles {
     pub main: PathBuf,
@@ -50,12 +52,18 @@ pub fn build_exe(root: &Path) -> io::Result<Vec<u8>> {
     let mut bytes = fs::read(env::current_exe()?)?;
     bytes.extend_from_slice(&files_bytes);
     bytes.extend((files_bytes.len() as u64).to_le_bytes());
+    bytes.extend(STAND_DATA_SIGNATURE);
     Ok(bytes)
 }
                                                                                                                            
 fn load_files() -> io::Result<StandFiles> {
     let mut bytes = fs::read(env::current_exe()?)?;
-    let files_len = u64::from_le_bytes(bytes.split_off(bytes.len() - 8).try_into().unwrap());
+    let mut data_block = bytes.split_off(bytes.len() - 16);
+    let signature = data_block.split_off(8);
+    if signature != STAND_DATA_SIGNATURE {
+        return Ok(StandFiles::default());
+    }
+    let files_len = u64::from_ne_bytes(data_block.try_into().unwrap());
     let start = bytes.len() - files_len as usize;
     let files: StandFiles = serde_json::from_slice(&bytes[start..])?;
     Ok(files)
```

**Comment by kaikalii** [comment]
I may have fixed this in f107f48dd7849db by implementing your suggestion.
However, it was never broken on my machine, so I cannot test if it worked.

**Comment by shnarazk** [comment]
@kaikalii 
fixed. Thanks!
